code-monkey/src/agents/base_agent.py

```diff
@@ -29,16 +19,6 @@ def get_system_prompt(self) -> str:
     def prepare_prompt(self, prompt: str) -> str:
         pass
 
-    def get_tools(self) -> List[Tool]:
-        if self.tools is None:
-            self.tools = [create_tool(spec) for spec in self.tool_specs]
-            self.tools_by_name = {tool.name: tool for tool in self.tools}
-        return self.tools
-
-    def get_tool(self, name: str) -> Dict[str, Tool]:
```

Comment:
Why was `get_tool` removed?

Reply:
not called by anything.  The tool linkage is handled by langchain, in `agent.py`:

```
        self.model = create_react_agent(
            ChatModel(
                model_name=model,
                extra_flags=extra_flags,
            ),
            self.tools,
            checkpointer=memory,
        )
```

actually ``BaseAgent.tools_by_name`, and the entirety of `tools/tools.py` can go as well.

Reply:
removed those extra things